### ISSUE TITLE
Better diffs for indent changes.

### ISSUE THREAD

**Issue by blainekasten** [dear-github/dear-github#91]
Currently, one of the hardest diffs to review, is when a bunch of code gets indented positively or negatively. It's hard to tell if the user actually made a change somewhere in the indent code because it all mostly looks the same. 

Here is prime example: https://gist.github.com/blainekasten/cfd44975df7c294c54e6/revisions

Hardly any code was changed, just de-indented.

It would help if we could see the actual code that changed, and then indent changes were hidden and noted. Something that looked more like this:

```diff
- removedThisFunction(function() {
+  de-indented containing content 2 spaces
- });

**Comment by mrdoob** [comment]
`?w=1` may be what you're after... https://github.com/blog/967-github-secrets

**Comment by blainekasten** [comment]
Wow. Completely am. That seems weird that it's a mostly hidden feature. It seems like it should be first-class. I don't think anyone benefits from seeing just tabbing changes.

**Comment by mrdoob** [comment]
I agree, they should really add a switch in the UI somewhere for this.

**Comment by Stargator** [comment]
:+1:

**Comment by Tanmayshetty** [comment]
:+1:

**Comment by jawshooah** [comment]
In the meantime, there is a [Chrome extension](https://chrome.google.com/webstore/detail/github-diff-whitespace/lhbcdehjihmbiafeodkfnbndleijnnhp?hl=en-US) that adds a button for you.

**Comment by danielbartsch** [comment]
Well there's a huge disadvantage with this method.

You cannot comment on lines in PRs.
I mean, that's clear, because for people who don't add `?w=1`, the diff might look different. But wouldn't a sufficient solution be to let the user decide which default diff representation is set?

In some (most) programming languages, the indent doesn't matter at all functionality wise. It would be cool to be able to set the default view (with or without whitespace diff) for a repo

**Comment by il--ya** [comment]
@mrdoob, There is a tickbox now to ignore whitespace on PR and commit diff pages (still missing on gist.github.com).
![image](https://user-images.githubusercontent.com/6630990/49321591-33a42280-f500-11e8-995d-c67a127f469b.png)
@blainekasten, the real problem is not the indentation per se, but the fact that Github's diff viewer doesn't try very hard to align the lines - it only looks for  identical lines to align the blocks, and then pads the smaller block with empty lines at the bottom. It doesn't try to align not-quite-identical lines. The only exception is made for the white space, but it is a manual option and it is quite blunt too as it ignores *all* whitespace regardless of how good the matching is overall.
What is required is a more flexible, fuzzy line alignment algorithm in general, which would try harder to align less than ideally matching lines and take into account how that affects matching of the nearby lines as well.

**Comment by timotheecour** [comment]
I've made a PR that shows exactly the problem with all options I currently know of: see https://github.com/timotheecour/Nim/pull/37
the actual diff contains a single large re-indented block; and a single re-indented line within that large block; none of the tools work well in this case: `git diff --color-moved-ws=allow-indentation-change --color-moved HEAD^` or https://github.com/timotheecour/Nim/pull/37/files?w=1  all have significant drawbacks, as explained in that PR

**Comment by sophiebits** [comment]
For what it's worth, Reviewable [handles @timotheecour's example](https://reviewable.io/reviews/timotheecour/Nim/37) well:

![screenshot of diff in Reviewable showing the entire re-indented block, where the one re-indented line can also be identified by color highlighting](https://user-images.githubusercontent.com/6820/183508178-07d8f524-9f4c-4264-927c-fd0caa045ec4.jpg)
